src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java

```diff
@@ -46,9 +46,13 @@ public static Builder builder() {
         return new AutoValue_ThreatIntelPluginConfiguration.Builder();
     }
 
-    @JsonIgnore
-    public boolean isOtxComplete() {
-        return otxApiKey() != null && !otxApiKey().isEmpty();
+    public static ThreatIntelPluginConfiguration defaults() {
```

Comment:
Server startup fails for me with the following stacktrace. (guice error) I don't have a `ThreadIntelPluginConfiguration` object in my cluster config.
I guess the `otxApiKey` property in `ThreadIntelPluginConfiguration` needs to be `@Nullable`.

```
Caused by: java.lang.IllegalStateException: Missing required properties: otxApiKey
	at org.graylog.plugins.threatintel.AutoValue_ThreatIntelPluginConfiguration$Builder.build(AutoValue_ThreatIntelPluginConfiguration.java:157)
	at org.graylog.plugins.threatintel.ThreatIntelPluginConfiguration.defaults(ThreatIntelPluginConfiguration.java:55)
	at org.graylog.plugins.threatintel.functions.global.AbstractGlobalLookupFunction.<init>(AbstractGlobalLookupFunction.java:28)
	at org.graylog.plugins.threatintel.functions.global.GlobalIpLookupFunction.<init>(GlobalIpLookupFunction.java:42)
	at org.graylog.plugins.threatintel.functions.global.GlobalIpLookupFunction$$FastClassByGuice$$cac62d3d.newInstance(<generated>)
	at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:111)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
```

Reply:
In addition, I think it would be good if we don't do a cluster config lookup (and thus MongoDB request) in the constructor of the `AbstractGlobalLookupFunction` class. (and others, if we do it somewhere else)

Reply:
Good catches, thanks!